src/org/zaproxy/zap/extension/api/RubyAPIGenerator.java

```diff
@@ -0,0 +1,227 @@
+			}
+			for (ApiElement action : imp.getApiActions()) {
+				this.generateRubyElement(action, imp.getPrefix(), "action", out);
+			}
+			for (ApiElement other : imp.getApiOthers()) {
+				this.generateRubyElement(other, imp.getPrefix(), "other", out);
+			}
+			out.write("  end\n");
+			out.write("end");
+			out.write("\n");
+		}
+	}
+
+	private void generateRubyElement(ApiElement element, String component, String type, Writer out) throws IOException {
+
+		boolean typeOther = type.equals("other");
+		boolean hasParams = (element.getMandatoryParamNames() != null && element.getMandatoryParamNames().size() > 0)
+				|| (element.getOptionalParamNames() != null && element.getOptionalParamNames().size() > 0);
+
+		// Add description if defined
+		String descTag = element.getDescriptionTag();
+		if (descTag == null) {
+			// This is the default, but it can be overridden by the getDescriptionTag method if required
+			descTag = component + ".api." + type + "." + element.getName();
+		}
+		try {
+			String desc = msgs.getString(descTag);
+			out.write("    # " + desc + "\n");
+			if (optional) {
+				out.write("    #\n");
+				out.write("    # " + OPTIONAL_MASSAGE + "\n");
+			}
+		} catch (Exception e) {
+			// Might not be set, so just print out the ones that are missing
+			System.out.println("No i18n for: " + descTag);
+			if (optional) {
+				out.write("    # " + OPTIONAL_MASSAGE + "\n");
+			}
+		}
+
+		// Method declaration
+		out.write("    def " + createMethodName(element.getName()));
+		out.write("(");
+		
+		// Iterate through and write out method arguments
+		ArrayList<String> args = new ArrayList<String>();
+		if (element.getMandatoryParamNames() != null) {
+			for (String param : element.getMandatoryParamNames()) {
+				args.add(param.toLowerCase());
+			}
+		}
+		if (element.getOptionalParamNames() != null) {
+			for (String param : element.getOptionalParamNames()) {
+				args.add(param.toLowerCase() + " = ''");
+			}
+		}
+		out.write(StringUtils.join(args, ", "));
+		out.write(")\n");
+
+		String method = "request";
+		if (typeOther) { method += "_other"; }
+
+		// Return method
+		out.write("      @zap." + method + "('" + component + "/" + type + "/" + element.getName() + "/'");
+
+		// Request parameters
+		if (hasParams) {
+			out.write(", {");
+			ArrayList<String> paramArgs = new ArrayList<String>();
+			
+			// Iterate through and write out mandatory request parameters
+			writeOutRequestParams(element.getMandatoryParamNames(), paramArgs, out);
+
+			// Iterate through and write out optional request parameters
+			writeOutRequestParams(element.getOptionalParamNames(), paramArgs, out);
+			
+			out.write(StringUtils.join(paramArgs, ", "));
+			out.write("}");
+		}
+
+		out.write(")");
+		out.write("\n");
+		out.write("    end\n");
+		out.write("\n");
+	}
+
+	// Writes the request parameters
+	private void writeOutRequestParams(List<String> elements, ArrayList<String> args, Writer out) throws IOException {
```

Comment:
The method should be renamed, it's not doing what the name says.
The parameter `out` should be removed (as well the declared thrown exception).